### PR TITLE
Limiting The Amount of Hak - Fixing Heights of Hak and Kewajiban

### PR DESCRIPTION
**Pull request by bryanjeshua**

### CODE REVIEW

**Comment by bryan273** [comment]
Hey, what a really nice work you've put into this PR! I'm genuinely impressed by the new `InputField` component with its character limit enforcement and inline counting feature. The way you've enhanced user feedback with toast notifications and enforced limits on rights and obligations is top-notch. Plus, the comprehensive test suites covering edge cases and interactions are a fantastic addition. Amazing job on the bug fixes and UI refinements like fixing the height of hak and kewajiban, truly well done!

However, I do have a couple of suggestions to polish this even further. First, regarding the test description issue in `inputfield.test.tsx` highlighted by @coderabbitai, aligning the test name with the actual behavior would avoid confusion. A small tweak like this could help:

```
it("should call onUpdate even when value is unchanged (blur)", async () => {
  render(<InputField {...defaultProps} />);

  const textarea = screen.getByTestId("mock-textarea");
  // Focus and blur without changing the value
  fireEvent.focus(textarea);
  fireEvent.blur(textarea);

  expect(mockOnUpdate).toHaveBeenCalledTimes(1);
});
```

Additionally, on the `partyId` type mismatch in `rights-and-obligation.tsx`, I agree with restricting it to `string` for TypeScript safety. A quick adjustment like below would prevent potential errors:

```
const handleUpdateRight = (
  partyId: string, // Restrict to string for TS compatibility
  idx: number,
  value: string
) => {
  // Existing logic...
};
```

These are minor tweaks to an otherwise stellar PR. Fantastic work again, and I'm excited to see this merged!

File: src/components/ui/form/inputfield.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  // If the parent state changes externally, sync the local state.
+  useEffect(() => {
+    setValue(initialValue);
+  }, [initialValue]);
+
+  const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    const newValue = e.target.value;
+    // Only keep the first maxLength characters
+    if (newValue.length <= maxLength) {
+      setValue(newValue);
```
Comment (eryawww) [inline src/components/ui/form/inputfield.tsx:36]: Hello bryan, great improvement!
However, I find few improvement that could be done to make the code more maintainable. 

These if else statement is unnecessary.
```python
const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
  setValue(e.target.value.substring(0, maxLength));
};
```
